dvc/repo/__init__.py

```diff
@@ -605,12 +605,19 @@ def site_cache_dir(self) -> str:
         finally:
             os.umask(umask)
 
+        # NOTE: Some number to change the generated token if none of the
+        # components were changed (useful to prevent newer dvc versions from
+        # using older broken cache). Please reset this back to 0 if other parts
+        # of the token components are changed.
+        salt = 0
```

Comment:
Should we version the path instead? Maybe I am thinking as an engineer, but I have a lot of `/var/tmp/dvc/repo` directories that I don't have a good idea what to keep and what to remove. :)

Reply:
@skshetry Also thought about that initially, but it worried that it will get unwieldy. But with the current setup, I guess we could reasonably do `{user}/{dvc_major}/{root_dir_token}/{salt}` or some combination like that. But still feels a bit unwieldy to put everything into that path, so maybe some compromise like `{user}/{dvc_major}/{root_token_with_salt}` would do.

But I totally agree that the issue of cleaning up site_cache_dir is not handled at all. At the same time, just nuking it pretty safe and might only cost some one-time computations.

Reply:
I guss the question really is: what kind of granularity would one want there? Cleanup by user seems reasonable. By dvc major version - I guess, but seems less useful.

Reply:
@skshetry Or maybe I didn't understand `version the path` correctly. Please elaborate.

Reply:
Sorry, I meant using something like `/var/tmp/dvc/repo/v{salt}/` that is unrelated to the dvc's major version. You can easily inspect by path, and figure out if it's supported or not. It might be useful in debugging/support too. :)

Reply:
oh, got it. Feels excessive to put the salt into the path, as it can be the same in multiple versions even if the token has changed. The best debugging/support approach seems to just nuke site_cache_dir.

Reply:
Goes a bit against the design of DVC, no? Transparent/inspectable/human-readable -ness.
But no strong opinion. Just wanted to suggest one alternative. 🙂

Reply:
@skshetry Yeah, but this is just cache and we can't encode everything into the token (e.g. root_dir), so we go the reverse route by just using a token here, but `dvc doctor` will tell you precisely the location of your cache.

I'm open to suggestions actually. Putting just salt seems odd, because it is just salt and we have more useful components like user/root/major version.

I'm happy to change it now or in the future.

Reply:
Another perk of the current approach is that it is easy to adjust to whatever you want, because it will be the same looking token no matter what, so no messing about with versioned paths and more complex logic like that.